src/app/server/Dnssd.cpp

```diff
@@ -145,6 +145,19 @@ CHIP_ERROR DnssdServer::SetEphemeralDiscriminator(Optional<uint16_t> discriminat
     return CHIP_NO_ERROR;
 }
 
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+template <class Derived>
+void DnssdServer::AddICDKeyToAdvertisement(Dnssd::BaseAdvertisingParams<Derived> & advParams)
```

Comment:
Wouldn't this be simpler as:
```suggestion
template <class AdvertisingParams>
void DnssdServer::AddICDKeyToAdvertisement(AdvertisingParams & advParams)
```
and then the callsites would not need to manually specialize the template.

Reply:
Yes, simpler/cleaner ! Thanks, applied.